AsyncDisplayKit/ASVideoNode.mm

```diff
@@ -153,43 +174,39 @@ - (void)setPlaceholderImagefromAsset:(AVAsset*)asset
     imageGenerator.appliesPreferredTrackTransform = YES;
     NSArray *times = @[[NSValue valueWithCMTime:CMTimeMake(0, 1)]];
     
-    [imageGenerator generateCGImagesAsynchronouslyForTimes:times completionHandler:^(CMTime requestedTime, CGImageRef  _Nullable image, CMTime actualTime, AVAssetImageGeneratorResult result, NSError * _Nullable error) {
+    [imageGenerator generateCGImagesAsynchronouslyForTimes:times completionHandler:^(CMTime requestedTime, CGImageRef _Nullable image, CMTime actualTime, AVAssetImageGeneratorResult result, NSError * _Nullable error) {
       
       ASDN::MutexLocker l(_videoLock);
       
       // Unfortunately it's not possible to generate a preview image for an HTTP live stream asset, so we'll give up here
       // http://stackoverflow.com/questions/32112205/m3u8-file-avassetimagegenerator-error
       if (image && _placeholderImageNode.image == nil) {
-        UIImage *theImage = [UIImage imageWithCGImage:image];
-        
-        if (!_placeholderImageNode) {
-          _placeholderImageNode = [[ASImageNode alloc] init];
-          _placeholderImageNode.layerBacked = YES;
-        }
-        
-        _placeholderImageNode.image = theImage;
-        
-        if ([_gravity isEqualToString:AVLayerVideoGravityResize]) {
-          _placeholderImageNode.contentMode = UIViewContentModeRedraw;
-        }
-        else if ([_gravity isEqualToString:AVLayerVideoGravityResizeAspect]) {
-          _placeholderImageNode.contentMode = UIViewContentModeScaleAspectFit;
-        }
-        else if ([_gravity isEqualToString:AVLayerVideoGravityResizeAspectFill]) {
-          _placeholderImageNode.contentMode = UIViewContentModeScaleAspectFill;
-        }
-        
-        dispatch_async(dispatch_get_main_queue(), ^{
-          ASDN::MutexLocker l(_videoLock);
-          
-          [self insertSubnode:_placeholderImageNode atIndex:0];
-          [self setNeedsLayout];
-        });
+        [self setPlaceholderImage:[UIImage imageWithCGImage:image]];
       }
     }];
   });
 }
 
+- (void)setPlaceholderImage:(UIImage *)image
```

Comment:
I like the idea. I'm more partial to it than my original idea of adding a [delegate method for thumbnail generation](https://github.com/ejensen/AsyncDisplayKit/compare/video-fixes...ejensen:video-placeholder-delegate). I'm curious how the relationship between videoGravity and contentMode would work.